src/utils/deployPeertube.ts

```diff
@@ -0,0 +1,146 @@
+import type { default as VM } from "../types/vm";
+import type { IProfile } from "../types/Profile";
+
+import { getUniqueName, selectGatewayNode } from "./gatewayHelpers";
+
+const { HTTPMessageBusClient } = window.configs?.client ?? {};
+const {
+  GridClient,
+  NetworkModel,
+  DiskModel,
+  MachineModel,
+  MachinesModel,
+  GatewayNameModel,
+} = window.configs?.grid3_client ?? {};
+
+export default async function deployPeertube(data: VM, profile: IProfile) {
+  const { envs, disks, ...base } = data;
+  let { name, flist, cpu, memory, entrypoint, network: nw } = base;
+  const { publicIp, planetary, nodeId, rootFsSize } = base;
+  const { mnemonics, storeSecret, networkEnv } = profile;
+
+  const http = new HTTPMessageBusClient(0, "");
+  const client = new GridClient(
+    networkEnv as any,
+    mnemonics,
+    storeSecret,
+    http,
+    undefined,
+    "tfkvstore" as any
+  );
+
+  await client.connect();
+
+  // Make sure the name is valid
+  name = await getUniqueName(client, name);
+
+  // Dynamically select node to deploy the gateway
+  let [gwNodeId, gwDomain] = await selectGatewayNode();
+  const domain = `${name}.${gwDomain}`;
+
+  // define a network
+  const network = new NetworkModel();
+  network.name = name + "Net";
+  network.ip_range = "10.1.0.0/16";
+
+  // deploy the peertube
+  await deployPeertubeVM(client, network, nodeId, name, domain);
+
+  // get the info of peertube deployment
+  const peertubeInfo = await getPeertubeInfo(client, name + "VMs");
+  const planetaryIP = peertubeInfo[0]["planetary"];
+
+  // deploy the gateway
+  await deployPrefixGateway(client, name, planetaryIP, gwNodeId);
+
+  // get the info of the deployed gateway
+  const gatewayInfo = await getGatewayInfo(client, name);
+  const gatewayDomain = gatewayInfo[0]["domain"];
+  return { domain, planetaryIP };
+}
+
+async function deployPeertubeVM(
+  client: any,
+  net: any,
+  nodeId: any,
+  name: string,
+  domain: string
+) {
+  // disk
+  const disk = new DiskModel();
+  disk.name = name + "Data";
+  disk.size = 10;
+  disk.mountpoint = "/data";
+
+  // vm specs
+  const vm = new MachineModel();
+  vm.name = name + "VM";
+  vm.node_id = nodeId;
+  vm.disks = [disk];
+  vm.public_ip = false;
+  vm.planetary = true;
+  vm.cpu = 3;
+  vm.memory = 1024 * 4;
+  vm.rootfs_size = 1;
+  vm.flist =
+    "https://hub.grid.tf/omarabdul3ziz.3bot/threefoldtech-peertube-v3.0.flist";
+  vm.entrypoint = "/usr/local/bin/entrypoint.sh";
+  vm.env = {
+    PEERTUBE_ADMIN_EMAIL: "support@incubid.com",
+    PEERTUBE_WEBSERVER_HOSTNAME: domain,
+    PEERTUBE_WEBSERVER_PORT: "443",
+    PEERTUBE_DB_SUFFIX: "_prod",
+    PEERTUBE_DB_USERNAME: "peertube",
+    PEERTUBE_DB_PASSWORD: "peertube",
+  };
+
+  // vms specs
+  const vms = new MachinesModel();
+  vms.name = name + "VMs";
+  vms.network = net;
+  vms.machines = [vm];
+
+  // deploy
+  window.configs.currentDeploymentStore.deploy("Peertube", name);
```

Comment:
can we use the generic function `deploy` here? it's define at [deploy.ts](https://github.com/threefoldtech/grid_weblets/blob/development/src/utils/deploy.ts) with a usage example at https://github.com/threefoldtech/grid_weblets/blob/d28491b3fb7bf0e83dbce1f8aa67f4536d7da001/src/utils/deployVM.ts#L32